### PR TITLE
Expose AWS instance metadata for @kubernetes

### PR DESCRIPTION
**Pull request by savingoyal**
Introduces optional metadata for tasks that execute on Kubernetes on EKS

### CODE REVIEW

File: metaflow/plugins/kubernetes/kubernetes_decorator.py [hunk 1]
```diff
@@ -329,6 +330,23 @@ def task_pre_step(
...
+
+            # Capture AWS instance identity metadata. This is best-effort only since
+            # access to this end-point might be blocked on AWS and not available
+            # for non-AWS deployments.
+            # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html
+            # TODO (savin): Introduce equivalent support for Microsoft Azure and
+            #               Google Cloud Platform
+            try:
+                instance_meta = requests.get(
+                    url="http://169.254.169.254/latest/dynamic/instance-identity/document"
```
Comment (obgibson) [inline metaflow/plugins/kubernetes/kubernetes_decorator.py:342]: Is this IP significant?

Comment (savingoyal) [reply metaflow/plugins/kubernetes/kubernetes_decorator.py:342]: Yes, it's a magic IP that hosts instance metadata for AWS, Azure and GCP

**Comment by obgibson** [comment]
To prevent uninformed questions like this, I suggest assigned the IP to a variable called magic_ip_that_hosts_metadata_for_aws_gcp_azure (is that pythonic?)

**Comment by savingoyal** [comment]
That's fair - I can push a different PR that addresses this comment.

**Pull request merged**